inheritance.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var Animal = function() {
+  this.stepsWalked = 0;
+  this.walk = function(steps) {
+    this.stepsWalked += steps;
+    console.log("i am walking ");
+  }
+  return this;
+};
+
+function Dog() {
+  Animal.call(this);
+}
+
+Dog.prototype = Object.create(Animal.prototype);
+Dog.prototype.constructor = Dog;
+
+Dog.prototype.wagTail = function() {
+  for (var i = 1; i <= 10; i++) {
+    setTimeout(function timer() {
+      console.log('waggle');
+    }, i * 3000);
```

Comment:
The requirements said to sleep 500 ms between each iteration; this loop sleeps (i * 3000) ms between each iteration.